[dv] Add GUI debug mode for Riviera

dariuszst opened the pull request:
Hi,
I added a missing riviera_gui_debug mode required after f88269d2658dcf0d9c75b7a507d0bda160751c47 update.
This is an initial version allowing running tests with dvsim.py script.

rswarbrick reviewed (commented):
This looks reasonable, but needs a bit of tidying up.
- For the TODO note, I think we should be able to explain what's going on without needing to call out to a merged PR.
- I think the two commits probably need squashing together.

dariuszst commented:
Thanks @rswarbrick  - I've updated my fork repository (removed this unnecessary commit) and I hope the change is now referenced correctly. I changed also the TODO note - is it OK now?

rswarbrick commented on hw/dv/tools/dvsim/riviera.hjson:
```diff
@@ -69,6 +69,13 @@
     }
+    {
+      name: riviera_gui_debug      
+      // TODO need to add options to setup Riviera gui debug environment and features (similary like in PR #24156 for Xelium)
```
I think this would be better if we were pointing "internally". Can't you just say something like "Similar to how we do it for Xcelium: see xcelium.hjson." ?

dariuszst replied:
Yes. It can be described in this way.

rswarbrick replied:
Probably just merge the changes together.

rswarbrick commented:
I've just squashed things together and dropped the trailing whitespace. Thanks for looking at this.
